framework/core/Project.pm

```diff
@@ -370,55 +385,57 @@ sub checkout_vid {
         }
     }
 
+    # Note: will skip both of these for bug mining, for two reasons, 1: it isnt necessary, 2: dont have depdencies yet
     # Fix test suite if necessary
     $self->fix_tests("${bid}f");
-
     # Write version-specific properties
-    $self->_write_props($vid, $work_dir);
+    $self->_write_props($vid, $is_bugmine);
 
     # Commit and tag the fixed program version
     $tag_name = Utils::tag_prefix($pid, $bid) . $TAG_FIXED;
-    $cmd = "cd $work_dir" .
+    $cmd = "cd $prog_root" .
            " && git add -A 2>&1" .
            " && git commit -a -m \"$tag_name\" 2>&1" .
            " && git tag $tag_name 2>&1";
     Utils::exec_cmd($cmd, "Initialize fixed program version")
             or confess("Couldn't tag fixed program version!");
 
-    # Apply patch to obtain buggy version
-    my $patch_dir = "$SCRIPT_DIR/projects/$pid/patches";
+    # bug mining uses a special patch dir because we make the patches on the fly
+    # if it exists we use it if not we look for the main db patches, if those dont exist we will fail
+    my $patch_dir =  (-d "$self->{_work_dir}/$pid/patches") ? "$self->{_work_dir}/$pid/patches" : "$SCRIPT_DIR/projects/$pid/patches";
     my $src_patch = "$patch_dir/${bid}.src.patch";
-    $self->apply_patch($work_dir, $src_patch) or return 0;
+    # Apply patch to obtain buggy version
+    $self->apply_patch($prog_root, $src_patch) or return 0;
 
     # Write program and version id of buggy program version to config file
-    Utils::write_config_file("$work_dir/$CONFIG", {$CONFIG_PID => $pid, $CONFIG_VID => "${bid}b"});
+    Utils::write_config_file("$prog_root/$CONFIG", {$CONFIG_PID => $pid, $CONFIG_VID => "${bid}b"});
 
     # Commit and tag the buggy program version
     $tag_name = Utils::tag_prefix($pid, $bid) . $TAG_BUGGY;
-    $cmd = "cd $work_dir" .
+    $cmd = "cd $prog_root" .
            " && git add -A 2>&1" .
            " && git commit -a -m \"$tag_name\" 2>&1" .
            " && git tag $tag_name 2>&1";
     Utils::exec_cmd($cmd, "Initialize buggy program version")
             or confess("Couldn't tag buggy program version!");
 
     # Checkout post-fix revision and apply unmodified diff to obtain the pre-fix revision
-    my $tmp_file = "$work_dir/.defects4j.diff";
-    $cmd = "cd $work_dir && git checkout " . Utils::tag_prefix($pid, $bid) . "$TAG_POST_FIX 2>&1";
+    my $tmp_file = "$prog_root/.defects4j.diff";
+    $cmd = "cd $prog_root && git checkout " . Utils::tag_prefix($pid, $bid) . "$TAG_POST_FIX 2>&1";
     `$cmd`; $?==0 or confess("Couldn't checkout $TAG_POST_FIX");
     my $rev1 = $self->lookup("${bid}f");
     my $rev2 = $self->lookup("${bid}b");
     # TODO: svn doesn't support diffing of binary files
     #       -> checkout and tag the pre-fix revision instead
-    $self->{_vcs}->export_diff($rev1, $rev2, $tmp_file);
-    $self->{_vcs}->apply_patch($work_dir, $tmp_file);
+    $self->export_diff($rev1, $rev2, $tmp_file, "src/");
```

Comment:
What's the hard-coded "src/" argument?

Reply:
I explicitly write src here to export the diff only from the src tree of the project, meaning `test/` and other directories will be excluded from any diff.

Reply:
The issue is that some projects store the sources in a differently named directory. For example, Chart's sources live in *source*. Also, isn't this diff supposed to include all the changes? This code computes and applies the diff from the post-fix revision to the pre-fix revision, which should be equivalent to checking out the pre-fix revision.

Reply:
Okay, I will have to adjust that so it export the diff of the entire project.